test/integration/scheduler_perf/util.go

```diff
@@ -640,3 +649,151 @@ func (tc *throughputCollector) collect() []DataItem {
 
 	return []DataItem{throughputSummary}
 }
+
+// memoryCollector collects memory usage metrics during the test
+type memoryCollector struct {
+	samples      []memorySample
+	resultLabels map[string]string
+	interval     time.Duration
+	stopCh       chan struct{}
+	mu           sync.RWMutex
+}
+
+type memorySample struct {
+	timestamp      time.Time
+	heapInuseMB    float64
+	goroutineCount float64
+}
+
+func newMemoryCollector(resultLabels map[string]string, interval time.Duration) *memoryCollector {
+	return &memoryCollector{
+		resultLabels: resultLabels,
+		interval:     interval,
+		stopCh:       make(chan struct{}),
+	}
+}
+
+func (mc *memoryCollector) init() error {
+	mc.samples = nil
+	return nil
+}
```

Comment:
There is no need to reset the samples. The new `memoryCollector` instance is created each time.
```suggestion
func (mc *memoryCollector) init() error {
	return nil
}
```